### PR TITLE
Stagev2

### PR DESCRIPTION
**Pull request by juliuscecilia33**

### CODE REVIEW

File: lib/screens/landing/login_screen.dart [hunk 1]
```diff
@@ -24,32 +39,107 @@ class _LoginScreenState extends State<LoginScreen> {
...
+      return;
+    }
+
+    setState(() {
+      isLoading = true;
+      searchResult = {};
+    });
+
+    const String apiUrl_email = 'http://localhost:8080/users/login';
+    const String apiUrl_username = 'http://localhost:8080/users/username/login';
```
Comment (juliuscecilia33) [inline lib/screens/landing/login_screen.dart:73]: why is there two calls to our backend just to login? should try to combine the logic into just one call for a user to login. probs need to discuss with @Dongchunhua1225 on this

File: lib/screens/landing/login_screen.dart [hunk 2]
```diff
@@ -24,32 +39,107 @@ class _LoginScreenState extends State<LoginScreen> {
...
 
     // Create a map with the collected data
-    final Map<String, dynamic> userData = {
+    final Map<String, dynamic> userData_email = {
       'userEmail': emailController.text,
       'userPassword': passwordController.text,
     };
-    print(userData);
+    final Map<String, dynamic> userData_username = {
```
Comment (juliuscecilia33) [inline lib/screens/landing/login_screen.dart:79]: if you're able to just get one call, there will be just one map

File: lib/screens/landing/login_screen.dart [hunk 3]
```diff
@@ -24,32 +39,107 @@ class _LoginScreenState extends State<LoginScreen> {
...
+
     try {
-      final response = await http.post(
-        Uri.parse(apiUrl),
-        body: json.encode(userData),
+      final response_email = http.post(
+        Uri.parse(apiUrl_email),
+        body: json.encode(userData_email),
         headers: {'Content-Type': 'application/json'},
       );
```
Comment (juliuscecilia33) [inline lib/screens/landing/login_screen.dart:89]: same here, if we're able to combine the logic on the backend for the user email and username to just one call, we would only have to send one http.post request to our backend which is more efficient

File: lib/screens/landing/login_screen.dart [hunk 4]
```diff
@@ -24,32 +39,107 @@ class _LoginScreenState extends State<LoginScreen> {
...
+          msg: "Login or password is wrong",
+          toastLength: Toast.LENGTH_SHORT,
+          gravity: ToastGravity.CENTER,
+          timeInSecForIosWeb: 1,
+          backgroundColor: Colors.grey,
+          textColor: Colors.white,
+        );
       }
     } catch (e) {
       print('Error occurred while sending data: $e');
```
Comment (juliuscecilia33) [inline lib/screens/landing/login_screen.dart:138]: Should add a error toast here

File: lib/screens/landing/login_screen.dart (hunk 1 above)
Comment (tangjingyu0812) [reply lib/screens/landing/login_screen.dart:73]: one is for username and the other one is for email, the user can log in if one of them works

File: lib/screens/landing/login_screen.dart (hunk 4 above)
Comment (tangjingyu0812) [reply lib/screens/landing/login_screen.dart:138]: I have already added it in the very end

File: lib/screens/landing/login_screen.dart (hunk 1 above)
Comment (juliuscecilia33) [reply lib/screens/landing/login_screen.dart:73]: ohh got it, so it will just be one api call anyways? since user can input either just username or email

**Pull request merged**